auslib/test/test_db.py

```diff
@@ -2422,6 +2422,32 @@ def testGetRulesMatchingQueryChannelGlobbing(self):
         rules = self._stripNullColumns(rules)
         self.assertEquals(rules, expected)
 
+        # To ensure length of ruleChannel is >=3
+        expected = [dict(rule_id=2, priority=100, backgroundRate=100, channel='r*', update_type='z', data_version=1)]
+        rules = self.rules.getRulesMatchingQuery(
+            dict(name='', product='', version='3.0', channel='releasetest',
+                 buildTarget='', buildID='', locale='', osVersion='', distribution='',
+                 distVersion='', headerArchitecture='',
+                 force=False, queryVersion=3,
+                 ),
+            fallbackChannel='releasetest'
+        )
+        rules = self._stripNullColumns(rules)
+        self.assertNotEquals(rules, expected)
+
+        # To ensure globbing at end only
+        expected = [dict(rule_id=2, priority=100, backgroundRate=100, channel='r*test*', update_type='z', data_version=1)]
```

Comment:
This is a good test case. Like the other one, you'll want to add this rule as part of setUp(). However, I think you'll want to expect no rules to match when you pass "channel='releasetest-cck-blah'" to getRulesMatchingQuery (because "r*test*" isn't a match). Does that make sense?